Ast/src/Lexer.cpp

```diff
@@ -160,6 +172,18 @@ std::string Lexeme::toString() const
     case QuotedString:
         return data ? format("\"%.*s\"", length, data) : "string";
 
+    case InterpStringBegin:
+        return data ? format("the beginning of an interpolated string (`%.*s`)", length, data) : "the beginning of an interpolated string";
+
+    case InterpStringMid:
+        return data ? format("the middle of an interpolated string (`%.*s`)", length, data) : "the middle of an interpolated string";
+
+    case InterpStringEnd:
+        return data ? format("the end of an interpolated string (`%.*s`)", length, data) : "the end of an interpolated string";
```

Comment:
These feels relatively lengthy. Could it be
```suggestion
    case InterpStringBegin:
        return data ? format("`%.*s{", length, data) : "the beginning of an interpolated string";

    case InterpStringMid:
        return data ? format("}%.*s{", length, data) : "the middle of an interpolated string";

    case InterpStringEnd:
        return data ? format("}%.*s`", length, data) : "the end of an interpolated string";
```